packages/@webex/plugin-meetings/test/unit/spec/meeting/muteState.js

```diff
@@ -509,3 +509,202 @@ describe('plugin-meetings', () => {
+    muteState = createMuteState(mediaType, meeting, direction, false);
+  }
+
+   const setupSpies = (muteState, mediaType) => {
+    setUnmuteAllowedSpy = mediaType === AUDIO ? meeting.mediaProperties.audioTrack?.setUnmuteAllowed : meeting.mediaProperties.videoTrack?.setUnmuteAllowed;
+    setServerMutedSpy = mediaType === AUDIO ? meeting.mediaProperties.audioTrack?.setServerMuted : meeting.mediaProperties.videoTrack?.setServerMuted;
+    setMutedSpy = mediaType === AUDIO ? meeting.mediaProperties.audioTrack?.setMuted : meeting.mediaProperties.videoTrack?.setMuted;
+  };
+
+  const tests = [
+    {mediaType: AUDIO, title: 'audio'},
+    {mediaType: VIDEO, title: 'video'}
+  ];
+
+  tests.forEach(({mediaType, title}) =>
+    describe(title, () => {
+
+      afterEach(() => {
+        sinon.restore();
+      });
+
+      describe('#handleLocalTrackChange',() => {
+
+         it('calls init()', async () => {
+          setup(mediaType);
+          const spy = sinon.spy(muteState, 'init');
+          muteState.handleLocalTrackChange(meeting);
+          assert.calledOnceWithExactly(spy, meeting);
+        });
+      });
+
+      describe('#init', () => {
+
+        afterEach(() => {
+          sinon.restore();
+        });
+
+        it('nothing goes bad when track is undefined', async () => {
+          setup(mediaType, false, false, false);
+          setupSpies(muteState, mediaType);
+  
+          muteState.init(meeting);
+          assert.isFalse(muteState.state.client.localMute);
+        });
+  
+        it('tests when track muted is true and remoteMuted is false', async () => {
+          setup(mediaType, false, true);
+          setupSpies(muteState, mediaType);
+  
+          muteState.init(meeting);
+  
+          assert.calledWith(setUnmuteAllowedSpy, muteState.state.server.unmuteAllowed);
+          assert.notCalled(setServerMutedSpy);
+          assert.calledOnce(MeetingUtil.remoteUpdateAudioVideo);
+          assert.isTrue(muteState.state.client.localMute);
+        });
+
+          
+        it('tests when track muted is false and remoteMuted is false', async () => {
+          setup(mediaType, false, false);
+          setupSpies(muteState, mediaType);
+          muteState.state.server.localMute = true;
+  
+          muteState.init(meeting);
+  
+          assert.calledWith(setUnmuteAllowedSpy, muteState.state.server.unmuteAllowed);
+          assert.notCalled(setServerMutedSpy);
+          assert.calledOnce(MeetingUtil.remoteUpdateAudioVideo);
+          assert.isFalse(muteState.state.client.localMute);
+        });
+  
+        it('tests when remoteMuted is true', async () => {
+          // testing that muteLocalTrack is called
+          setup(mediaType, true);
+          setupSpies(muteState, mediaType);
+  
+          muteState.init(meeting);
+  
+          const serverMutedSpyCall1 = setServerMutedSpy.getCall(0);
+          const serverMutedSpyCall2 = setServerMutedSpy.getCall(1);
```

Comment:
why are we expecting 2 calls to setServerMuted? the init() should trigger just 1 call to it.
I've checked this and it's because the first call is triggered from init() called from createMuteState() that's called by setup(), so I think we should call setServerMutedSpy.resetHistory() in setup() after calling createMuteState() (or inside setupSpies()) so that we have a clean slate for the test. We should probably do it for all the spies.